components/server/src/projects/projects-service.ts

```diff
@@ -442,6 +442,16 @@ export class ProjectsService {
             throw new ApplicationError(ErrorCodes.NOT_FOUND, `Project ${partialProject.id} not found.`);
         }
 
+        // In case we are altering the "enableDockerdAuthentication": require org write_settings permission, so users
+        // can't inadvertently share credentials an owner configured
+        if (
+            partialProject?.settings?.enableDockerdAuthentication !==
+                existingProject.settings?.enableDockerdAuthentication ||
```

Comment:
I think we want this to be an AND, because as it currently stands, we would do a permission check if the user wasn't changing anything (with the setting already enabled).

```suggestion
                existingProject.settings?.enableDockerdAuthentication &&
```